### PR TITLE
Feat/GitHub contributors

### PR DESCRIPTION
**Pull request by xavier-zenika**
## Summary by CodeRabbit

- **New Features**
  - Added lazy loading functionality for images and iframes.
  - Introduced `GithubContributors` component to display GitHub contributors.
  - Implemented `ButtonBadge` component for customizable button badges.
  - Created `ExternalLink` component for rendering external links.

- **Bug Fixes**
  - Corrected typos and formatting issues in middleware documentation.
  - Fixed the array structure in the social links configuration.

- **Enhancements**
  - Updated sorting functions for tutorial links.
  - Enhanced API components with new props and type annotations.
  - Improved test cases for better coverage.
  - Corrected syntax error in `tailwind.preset.ts`.

- **Dependencies**
  - Added `axios` dependency.
  - Updated versions for `vitepress`, `vitest`, and `vue`.

- **Documentation**
  - Updated migration steps for clarity.

- **Configuration**
  - Adjusted code coverage thresholds in `vitest.config.ts`.
  - Removed `"jsdom"` from `tsconfig.json` types.

### CODE REVIEW

File: docs/.vitepress/theme/organisms/github-contributors/GithubContributors.vue [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+  setup(props) {
+    const contributors = ref([]);
+
+    const theme = useThemeConfig();
+    const {repo, githubProxyUrl, docsRepo = repo} = theme.value;
+
+    const {fetchContributors} = useGithubContributors(githubProxyUrl || docsRepo);
+
+    onMounted(async () => {
```
Comment (Romakita) [inline docs/.vitepress/theme/organisms/github-contributors/GithubContributors.vue:81]: wow Top :D
I just learned something ;)

File: readme.md [hunk 2]
```diff
@@ -62,16 +62,17 @@ So right now, we migrate the content without all custom components.
...
 
-1) Ask me to be added to the migration team (it gives you access to this repo)
-1) Go to this link to pick and assign you a task: https://github.com/tsedio/tsed-website/issues/14
-2) Start PR
-3) Get the legacy content from https://github.com/tsedio/tsed/tree/production/docs
-4) Review the content and migrate syntax that must me converted (see [syntax migration](#syntax-migration))
-5) Submit PR
+1. Ask me to be added to the migration team (it gives you access to this repo)
+1. Go to this link to pick and assign you a task: https://github.com/tsedio/tsed-website/issues/14
+1. Start PR
```
Comment (Romakita) [inline readme.md:67]: hum it should be 1., 2., 3. right ?

**Review by Romakita — CHANGES_REQUESTED** [review]

File: readme.md (hunk 2 above)
Comment (xavier-zenika) [reply readme.md]: Lol! Just fixed this

**Comment by Romakita** [comment]
@coderabbitai review

File: docs/.vitepress/theme/organisms/github-contributors/GithubContributors.vue [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+
+onMounted(async () => {
+  let fetchedContributorsRef = await fetchContributors();
+  let fetchedContributors = fetchedContributorsRef?.value;
+
+  if (fetchedContributors) {
+    // filtering logic (optional??)
+    if (props.users.length) {
+      fetchedContributors = fetchedContributors?.filter((contributor) => {
+        console.log(contributor, props.users);
```
Comment (Romakita) [inline docs/.vitepress/theme/organisms/github-contributors/GithubContributors.vue:57]: remove the console.log

File: docs/.vitepress/theme/organisms/github-contributors/GithubContributors.vue [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+        return props.users[0] === contributor.login;
+      });
+    }
+    contributors.value = fetchedContributors;
+  } else {
+    console.error("Error fetching contributors");
+  }
+
+  console.log("GITHUB_CONTRIBUTOR inside mounted", fetchedContributors);
```
Comment (Romakita) [inline docs/.vitepress/theme/organisms/github-contributors/GithubContributors.vue:66]: idem

File: docs/.vitepress/theme/molecules/contributors/ContributorItems.vue [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+  bgColor: string;
+  color: string;
+  blur: number;
+  textSize: string;
+  shadow: string;
+  padding: string | number;
+  innerPadding: string | number;
+  fontWeight: string;
+  liClass: string;
+  testUser: string;
```
Comment (Romakita) [inline docs/.vitepress/theme/molecules/contributors/ContributorItems.vue:35]: testUser ?

**Review by Romakita — DISMISSED** [review]

**Review by Romakita — APPROVED** [review]

**Pull request merged**